model/src/main/kotlin/Identifier.kt

```diff
@@ -122,10 +124,10 @@ data class Identifier(
     /**
      * Create a path based on the properties of the [Identifier]. All properties are encoded using [encodeOrUnknown].
      */
-    fun toPath() = listOf(provider, namespace, name, version).joinToString("/") { it.encodeOrUnknown() }
+    fun toPath() = list.joinToString("/") { it.encodeOrUnknown() }
 
     // TODO: Consider using a PURL here, see https://github.com/package-url/purl-spec#purl.
-    override fun toString() = "$provider:$namespace:$name:$version"
+    override fun toString() = list.joinToString(":") { it.trim().filterNot { it < ' ' } }
```

Comment:
Is it sufficient to do this here? We probably print components of the identifier in other places, too, so doing it in the constructor already would make sure it is always clean.

Reply:
True. I was hoping I could use property delegation with something like a `TrimmedSingleLine` class, but unfortunately [it's not possible to use delegated properties in the primary constructor of data classes](https://discuss.kotlinlang.org/t/using-delegated-properties-in-data-classes/6858).